pkg/controller/clusterpool/collections.go

```diff
@@ -584,6 +599,252 @@ func (cds *cdCollection) Delete(c client.Client, cdName string) error {
 	return nil
 }
 
+type cdcCollection struct {
+	// Unclaimed by any cluster pool CD any not broken
+	unassigned []*hivev1.ClusterDeploymentCustomization
+	// Listed in the cluster pool inventory but not found
+	missing []string
+	// Used by some cluster deployment
+	reserved []*hivev1.ClusterDeploymentCustomization
+	// Last Cluster Deployment failed on provision
+	cloud []*hivev1.ClusterDeploymentCustomization
+	// Failed to apply patches for this cluster pool
+	syntax []*hivev1.ClusterDeploymentCustomization
+	// All CDCs in this pool
+	byCDCName map[string]*hivev1.ClusterDeploymentCustomization
+}
+
+// getAllCustomizationsForPool is the constructor for a cdcCollection for all of the
+// ClusterDeploymentCustomizations that are related to specified pool.
+func getAllCustomizationsForPool(c client.Client, pool *hivev1.ClusterPool, logger log.FieldLogger) (*cdcCollection, error) {
+	if pool.Spec.Inventory == nil {
+		return nil, nil
+	}
+	cdcList := &hivev1.ClusterDeploymentCustomizationList{}
+	if err := c.List(
+		context.Background(), cdcList,
+		client.MatchingFields{claimClusterPoolIndex: pool.Name},
```

Comment:
Right, so I think I mentioned this elsewhere, but we're not yet constructing this index, so I wouldn't expect this to do anything. (Is it working?) Anyway, I think `InNamespace` is adequate, and we can filter out the ones that belong to this pool in the loops below.